homeassistant/components/generic/config_flow.py

```diff
@@ -515,3 +537,97 @@ async def get(self, request: web.Request, flow_id: str) -> web.Response:
             CAMERA_IMAGE_TIMEOUT,
         )
         return web.Response(body=image.content, content_type=image.content_type)
+
+
+class PreviewStream:
+    """A wrapper around the stream object to automatically close unused streams."""
+
+    def __init__(self, stream: Stream) -> None:
+        """Initialize the object."""
+        self.stream = stream
+        self._deferred_stop = None
+
+    async def start(self, timeout=600):
+        """Start the stream with a timeout."""
+
+        async def _timeout() -> None:
+            _LOGGER.debug("Starting preview stream with timeout %ss", timeout)
+            await asyncio.sleep(timeout)
```

Comment:
Could we use the default timeout built into the stream provider? My impression is that each stream has an idle timeout that can be used and will be renewed every time it is accessed. If that works here then perhaps we don't need additional timeout logic on top and can just rely on that to close the stream if things get abandoned.

Reply:
Additionally:  If you (1) move the code that does `stream.add_provider` into this function and  (2) always add a call to start when the websocket is opened then it can also be defensive against the user taking too long between the stream test step and the preview step. It can reuse the existing stream or start a new one if its not active.

Reply:
Ok, I think I get what you mean.

But there is no user controlled delay between the stream test and the preview step. I.e. if the stream passes the test, and all other tests are ok, the config flow immediately advances to the next page where the preview is shown (the user has no control of this timing).

I've done some basic tests with moving the PreviewStream wrapper altogether and just using Stream. That works too, should I do this.

Note that I wasn't able to write a test to verify that the stream had really timed out.

Reply:
OK, sounds good.

Regarding the delay between steps, i'm trying to say that calling `stream.add_provider` is idempotent and it can be called before the stream is shown to add a new idle timer if the stream already timed out because the delay between steps was too long.

I think its reasonable to not have a test verify this and trust the stream component will handle the timeout, so that sounds fine.

Reply:
I was still expecting this PR would remove this timeout behavior and rely on the existing stream timeout.

Reply:
PreviewStream wrapper is now dropped in favour of the built in timeout. The pytest should be triggering a timeout but I wasn't able to find a way to verify that it actually happened. The coverage hits 100% though.